### ISSUE TITLE
Enable running test scenarios in .NET containers without SDK installed

### ISSUE THREAD

**Issue by mthalman** [dotnet/sdk#26107]
For the [official .NET container images](https://github.com/dotnet/dotnet-docker), there's a need to be able to run product test scenarios within those containers. This will allow us to validate whether the container environment is configured appropriately for .NET app execution. In the case of distroless containers,  there's no shell available within the container to execute commands so the environment is inherently constrained (you could think of it like a device). With such an environment test management/execution would need to be done on the host which would control the single executions of an app in the distroless container.

The request here is to have the infrastructure available to execute product test scenarios in this distroless container environment.

**Comment by mthalman** [comment]
This scenario can be expressed more broadly to cover any scenario where you want to test any .NET application container and not have the SDK installed in it. The desire would be to have `dotnet test` run from the host and target an individual test case against a container instance. The test would run in the container environment I intend to deploy and so I would not want to have installed any additional test dependencies or test runner.

cc @agocke

**Comment by nohwnd** [comment]
@mthalman I am not sure if I am getting your description correctly 🙂 There are at least 2 ways to do this.

Here are the terms I will use below so we can hopefully understand each other better: There are 2 main components of Test Platform involved. There is vstest.console/dotnet test which orchestrates the run (**runner**) and prints the output and summary, and testhost that executes the actual test methods (**testhost**), but does not print any output and can't run on its own.

Then there is a **container** with an **image.** And an **entrypoint** to that container.

---

### Scenario 1: Container is host 

From your description I am assuming you want the runner to run outside of the container on some computer (be it the docker host, or some other computer), and orchestrate startup of docker containers. The containers would be based on dotnet/runtime images, with additional layer that has the SomeTests.dll and its dependencies that it gets from Microsoft.NET.Test.SDK.

There would be an entrypoint that would start the testhost application and listen for incoming test requests. 

The runner would start the container, tell it to run a given test, or all tests in the dll. And then then it would exit the container. 

The runner would not be responsible for building the image.

---
### Scenario 2: Container is runner and host

The runner runs inside of the container, there is layer that has the runner, and a layer that has SomeTests.dll and the dependencies it gets from Microsoft.NET.Test.SDK. 

The runner is the entrypoint of the container, and success or failure are communicated via exit codes (and additional trx output you can collect via mounts).

---

From your description I understand you want scenario 1. But Scenario 2 is pretty close to how remote test execution in containers works today in VS. And you can build it. And you can also use tools like `--blame-crash` to collect dumps of crashes etc.

**Comment by nohwnd** [comment]
2) it does not, the runner piece is shipped in SDK, but also as standalone nuget packages, so you can just extract them to a layer in your image and it will work as long as you have dotnet runtime available. 

dotnet/sdk shipment with .NET vstest.console: https://www.nuget.org/packages/Microsoft.TestPlatform.CLI/17.4.0-preview-20220726-02

**Comment by Youssef1313** [comment]
This should hopefully be possible already today with Microsoft.Testing.Platform. Is it necessary to do anything here for VSTest? Or is the MTP support sufficient and we can go ahead and close this?

**Comment by nohwnd** [comment]
Yes, running in container is possible, we have example of this in MTP. https://youtu.be/9Jz47Ze9LOI?si=ATBh8ku4FZGH7v7k&t=750

In vstest you could use the shipment linked above, which also does not require sdk.
